### PR TITLE
Bug 1152500 - Fix start/stop recording race conditions.

### PR DESCRIPTION
**Pull request by aosmond**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=be3d834caa68b63ede1d9504d39837633f8ba4bb

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=01ed13bcaf1b74f7993504b1af951e5ea32d7cf2

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0bdf3d6f85ac5033f82418979ebd0ded1fd247f2

File: apps/camera/js/lib/camera/camera.js [hunk 1]
```diff
@@ -939,41 +980,30 @@ Camera.prototype.stopRecording = function() {
...
-
+  var self = this;
+  var elapsedTime = self.get('videoElapsed');
+  var filepath;
+  var req;
+  if (recorded) {
     // Re-fetch the blob from storage
-    var req = storage.get(filepath);
-    req.onerror = self.onRecordingError;
+    var storage = this.storage.video;
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: nit: `storage` should be declared outside of the `if` block like `filepath` and `req` above.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9efac9272cecd8535580bfac2206ebd5870dffb3

**Pull request merged**